### PR TITLE
feat: upgrade packages and add package json commands

### PR DESCRIPTION
**Pull request by abnersajr**
## Summary

The project was on stale mode and with the recent drama on TFT we decided to bring back to life the POE Trading Discord.
Nothing better than do a clean and upgrade the old packages.

This PR updates all packages and lock exact versions, with the exception of `Discord.js`. This will be upgraded in their own PR.

SWC is also introduced to a faster build time.
Some comands to help the development:

```sh
yarn migration:run
yarn dev:watch # runs both swc and ts-node on watch mode
yarn build:watch # only for build + watch, used in the `dev:watch` command
```

So far Bot is working properly with these recent updates.

## Screenshots

Self-Vouch|Vouch + Check
-----|-----
self-vouch|vouch

### CODE REVIEW

**Review by multiplicitypoe — COMMENTED** [review]
So far looks good! Was the bot still able to build and run locally?

**Comment by abnersajr** [comment]
> So far looks good! Was the bot still able to build and run locally?

This PR still in Draft, not finished. Opened just for tracking purpose.
I will ask review when it's ready.

File: README.md [hunk 1]
```diff
@@ -2,14 +2,16 @@
...
    1. For local builds, use the `./dist` prefix to your migrations, i.e. `TYPEORM_ENTITIES=./dist/entities/*.js`
+   1. For development mode, use the `./src` prefix to migration and entities, with the `ts` file extension.
    1. For Docker builds, use root, i.e. `TYPEORM_ENTITIES=entities/*.js`
 1. Build the project
    1. Local: `yarn` then `yarn build`
    1. Docker: `docker build -t <your_tag> .`
 1. Create a Postgres database
 1. Run the migration scripts
-   1. Local: `./node_modules/.bin/typeorm migration:run`
+   1. Local: `yanr migration:run`
```
Comment (multiplicitypoe) [inline README.md]: nit:

```suggestion
   1. Local: `yarn migration:run`
```

File: src/bot.ts [hunk 2]
```diff
@@ -1,68 +1,63 @@
...
+  discordClient.on('message', (message: Message) => void checkVouchCommandHandler.handle(message));
 
-  client.on('messageDelete', (message: Message | PartialMessage) => void deleteVouchEvent.handle(message));
+  discordClient.on('messageDelete', (message: Message | PartialMessage) => void deleteVouchEvent.handle(message));
 }
 
 function ensureEnvironmentVariablesAreSet() {
   const requiredEnvVars = [
     'DISCORD_TOKEN',
-    'PREFIX',
```
Comment (multiplicitypoe) [inline src/bot.ts:65]: question: As far as you can tell, this isn't used anywhere?

**Review by multiplicitypoe — APPROVED** [review]
Looks like a great modernizing PR! Glad to see the formatting all cleaned up, dependencies bumped and the new data source wrapper

File: src/bot.ts (hunk 2 above)
Comment (abnersajr) [reply src/bot.ts:65]: Looking back through the history, I found that in the begin this was kind of the Prefix for the commands. It was removed in a certain point with a code refactor and it's not used anymore.

**Pull request merged**